### PR TITLE
feat: support firing of multiple rules

### PR DESCRIPTION
**Pull request by mkanoor**
https://issues.redhat.com/browse/AAP-9755
https://issues.redhat.com/browse/AAP-13131

Drools has always supported firing of multiple rules when an event or a fact comes in. Durable rules only supported this behavior for facts but not for events. This PR adds a feature to a ruleset to indicate if it should match_multiple rules. The default is false

https://github.com/ansible/ansible-rulebook/assets/6452699/20daa5aa-75c7-4c63-93f0-01678ba4c986

### CODE REVIEW

**Comment by Alex-Izquierdo** [comment]
This PR doesn't require an upgrade of drools-jpy? I believe it depends of some PR to be merged in the drools-integration project.

**Review by Alex-Izquierdo — CHANGES_REQUESTED** [review]
Release notes update missing.

**Comment by mkanoor** [comment]
@Alex-Izquierdo this was added in drools-jpy via https://issues.redhat.com/browse/AAP-13076, the current drools_jpy already has this change, we don't need a new drools-jpy

**Review by Alex-Izquierdo — DISMISSED** [review]
I really think it should have some e2e test as well, regardless of that LGTM

File: ansible_rulebook/rule_types.py [hunk 1]
```diff
@@ -78,6 +78,7 @@ class RuleSet(NamedTuple):
     gather_facts: bool
     uuid: Optional[str] = None
     default_events_ttl: Optional[str] = None
+    match_multiple_rules: Optional[bool] = False
```
Comment (Alex-Izquierdo) [inline ansible_rulebook/rule_types.py:81]: Minor: type hint is wrong. If the value is not nullable, it should not be optional.
```suggestion
    match_multiple_rules: bool = False
```

**Comment by ttuffin** [comment]
> I really think it should have some e2e test as well, regardless of that LGTM

+1. It would be great if any new features for rulebook will have coverage added to the e2e tests.

**Comment by mkanoor** [comment]
@Alex-Izquierdo @ttuffin I have added the e2e tests

**Comment by benthomasson** [comment]
@mkanoor this exposes a bug in the Rule Audit view assumptions.   This page is rule-focused and requires hunting for the event that caused the rule to fire instead of being event centric and showing what rules and actions fired from that event.

**Comment by benthomasson** [comment]
We should hold off on merging this until we have a meeting to discuss the Rule Audit view.

**Review by benthomasson — APPROVED** [review]

**Pull request merged**